lib/mutant/isolation.rb

```diff
@@ -33,30 +37,41 @@ module Fork
       #   if block terminates abnormal
       #
       # rubocop:disable MethodLength
-      def self.call(&block)
-        IO.pipe(binmode: true) do |reader, writer|
+      def call(&block)
+        io.pipe(binmode: true) do |reader, writer|
           writer.binmode
           begin
-            pid = Process.fork do
-              File.open(File::NULL, File::WRONLY) do |file|
-                $stderr.reopen(file)
-                reader.close
-                writer.write(Marshal.dump(block.call))
-                writer.close
-              end
+            pid = process.fork do
+              child(reader, writer, &block)
             end
 
             writer.close
-            Marshal.load(reader.read)
+            marshal.load(reader.read)
           ensure
-            Process.waitpid(pid) if pid
+            process.waitpid(pid) if pid
           end
         end
       rescue => exception
         raise Error, exception
       end
 
-    end # Fork
+      # Handle child process
+      #
+      # @param [IO] reader
+      # @param [IO] writer
+      #
+      # @return [undefined]
+      def child(reader, writer)
+        reader.close
 
+        devnull.call do |null|
+          stderr.reopen(null)
+          stdout.reopen(null)
+          writer.write(marshal.dump(yield))
```

Comment:
Interesting. But sadly there is no documentation for:

* How does it handle partial writes
* Does it use rubies buffering or not

So I'll change this one to an explicit syswrite instead. Also I suspect they write small amounts of data per object traversed (why would you just ofter such an interface if not) for our use case a single write will be more effective.